### PR TITLE
Reports rest

### PR DESCRIPTION
**Pull request by thescouser89**
[DA-3] Creates the initial REST endpoints for reports generator.

I've made some modifications to the pom.xml to change the packaging from ejb to war to get the REST endpoints activated. It seems to work but I have no clue what I'm doing :)

Endpoints are: `/reports-rest/generator/scm` and `/reports-rest/generator/gav`

I also don't know why @jomrazek commits are appearing in this PR.

### CODE REVIEW

**Comment by janinko** [comment]
If you rebase the reports_rest branch on master and force-push the branch jomrazek's commits should dissapear.

**Comment by thescouser89** [comment]
@janinko I'm gonna wait for the BW PR from @jomrazek  to be merged before force-pushing my change. Since our commit changes the same pom.xml, might as well wait.

I'll cherry-pick only my REST endpoint changes and force-push that.

**Comment by jomrazek** [comment]
after rebase :+1:

**Comment by thescouser89** [comment]
Rebased!

File: reports-rest/src/main/java/org/jboss/da/reports/rest/Generator.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.jboss.da.reports.rest;
+
+import org.jboss.da.reports.rest.model.GAVRequest;
+import org.jboss.da.reports.rest.model.Report;
+import org.jboss.da.reports.rest.model.SCMRequest;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+
```
Comment (jomrazek) [inline reports-rest/src/main/java/org/jboss/da/reports/rest/Generator.java]: Can this be interface when we don't have implementation yet?

Comment (jbartece) [inline reports-rest/src/main/java/org/jboss/da/reports/rest/Generator.java]: +1 for interface approach ;-)

**Comment by jbartece** [comment]
add to whitelist

**Comment by jbartece** [comment]
retest this please

**Comment by jbartece** [comment]
retest this please

**Comment by thescouser89** [comment]
I hope this failed due to a configuration error! :flushed:

**Comment by thescouser89** [comment]
retest this please

**Comment by thescouser89** [comment]
:pray:

**Pull request merged**